frontend/tests/unit/test_source_generator.py

```diff
@@ -0,0 +1,149 @@
+import os
+import pytest
+import platform
+import glob
+import re
+from filecmp import dircmp
+import sh
+from frontend_benchmark.user_callgraph import Callgraph
+from frontend_benchmark.source_generator import SourceGenerator
+
+
+@pytest.fixture
+def rootdir():
+    return os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                        os.path.pardir)
+
+
+@pytest.fixture
+def resources(rootdir):
+    return os.path.join(rootdir, 'resources')
+
+
+def test_branch_indirect_call(resources, tmpdir):
+    test_file = os.path.join(resources, 'branch_indirect_call.pbtext')
+    cfg = Callgraph.from_proto(test_file)
+    source_gen = SourceGenerator(tmpdir, cfg)
+    source_gen.write_files()
+    output = compile_and_return_output(
+        tmpdir, 'function_2', switch_processor(r'callq\s+\*', r'\s+blr\s+x\d+'))
```

Comment:
Put these assembly sequences as module-level constants for better readability (something like X86_INDIRECT_CALL/ARM64_INDIRECT_CALL).